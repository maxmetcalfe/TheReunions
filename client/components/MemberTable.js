import React, { Component } from "react";
import PropTypes from "prop-types";
import MemberCell from './MemberCell';

class MemberTable extends Component {

  render() {
    return (
      <table className="member-table">
        <tbody>
          {Object.keys(this.props.summaryCounts).map((name) => (
              <MemberCell key={name}
                member={name}
                counts={this.props.summaryCounts[name]}
                reunions={this.props.reunionsForMember[name]}
                selection={this.props.selection}
                setSelection={this.props.setSelection}
              />
          ))}
        </tbody>
      </table>
    );
  }
}

MemberTable.propTypes = {
  selection: PropTypes.object,
  reunionsForMember: PropTypes.object.isRequired,
  setSelection: PropTypes.func,
  summaryCounts: PropTypes.object.isRequired
};

export default MemberTable;
