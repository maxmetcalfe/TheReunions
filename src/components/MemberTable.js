import React from 'react';
import PropTypes from 'prop-types';

function App(props) {
  console.log(props);
  return (
    <table id="member-table">
      <tbody>
        {props.members.map((item)=>{
           return(
             <tr>item</tr>
           );
        })}
      </tbody>
    </table>
  );
};

App.propTypes = {
  members: PropTypes.string,
};

export default App;
