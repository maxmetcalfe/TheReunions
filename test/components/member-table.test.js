import React from "react"
import renderer from "react-test-renderer"
import MemberTable from "../../client/components/MemberTable"

const summaryCounts = {
    "j": { "four": 5, "three": 4, "two": 4 },
    "m": { "four": 5, "three": 3, "two": 3 },
    "d": { "four": 5, "three": 2, "two": 2 },
    "g": { "four": 5, "three": 1, "two": 1 },
}

const reunionsForMember = {
  "d": [
    { "category": "three", "element_id": "june2018THREE-Boston-1", "location": "Boston ", "name": "June 2018" },
    { "category": "two", "element_id": "july2018TWO-Boston-1", "location": "Boston ", "name": "July 2018" },
    { "category": "one", "element_id": "august2018ONE-Boston-1", "location": "Boston ", "name": "August 2018" }
  ],
  "g": [
    { "category": "three", "element_id": "june2018THREE-Boston-1", "location": "Boston ", "name": "June 2018" },
    { "category": "two", "element_id": "july2018TWO-Boston-1", "location": "Boston ", "name": "July 2018" },
    { "category": "one", "element_id": "august2018ONE-Boston-1", "location": "Boston ", "name": "August 2018" }
  ],
  "j": [
    { "category": "three", "element_id": "june2018THREE-Boston-1", "location": "Boston ", "name": "June 2018" },
    { "category": "two", "element_id": "july2018TWO-Boston-1", "location": "Boston ", "name": "July 2018" },
    { "category": "one", "element_id": "august2018ONE-Boston-1", "location": "Boston ", "name": "August 2018" }
  ],
  "m": [
    { "category": "three", "element_id": "june2018THREE-Boston-1", "location": "Boston ", "name": "June 2018" },
    { "category": "two", "element_id": "july2018TWO-Boston-1", "location": "Boston ", "name": "July 2018" },
    { "category": "three", "element_id": "august2014THREE-Boston-1", "location": "Boston ", "name": "August 2014" }
  ]
}

test("MemberTable", () => {
  let component = renderer.create(
    <MemberTable summaryCounts={summaryCounts} reunionsForMember={reunionsForMember} ></MemberTable>
  )

  expect(component).toMatchSnapshot()
})
