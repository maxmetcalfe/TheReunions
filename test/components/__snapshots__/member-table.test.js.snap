// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberTable 1`] = `
<table
  className="member-table"
>
  <tbody>
    <tr>
      <td>
        <div
          className="member-icon"
          id="button"
          onClick={[Function]}
        >
          J
        </div>
        <div
          className="coin-box"
        >
          <div
            className="coin four"
            onClick={[Function]}
          >
            5
          </div>
          <div
            className="coin three"
            onClick={[Function]}
          >
            4
          </div>
          <div
            className="coin two"
            onClick={[Function]}
          >
            4
          </div>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="member-icon"
          id="button"
          onClick={[Function]}
        >
          M
        </div>
        <div
          className="coin-box"
        >
          <div
            className="coin four"
            onClick={[Function]}
          >
            5
          </div>
          <div
            className="coin three"
            onClick={[Function]}
          >
            3
          </div>
          <div
            className="coin two"
            onClick={[Function]}
          >
            3
          </div>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="member-icon"
          id="button"
          onClick={[Function]}
        >
          D
        </div>
        <div
          className="coin-box"
        >
          <div
            className="coin four"
            onClick={[Function]}
          >
            5
          </div>
          <div
            className="coin three"
            onClick={[Function]}
          >
            2
          </div>
          <div
            className="coin two"
            onClick={[Function]}
          >
            2
          </div>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        <div
          className="member-icon"
          id="button"
          onClick={[Function]}
        >
          G
        </div>
        <div
          className="coin-box"
        >
          <div
            className="coin four"
            onClick={[Function]}
          >
            5
          </div>
          <div
            className="coin three"
            onClick={[Function]}
          >
            1
          </div>
          <div
            className="coin two"
            onClick={[Function]}
          >
            1
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
